noxfile.py

```diff
@@ -185,23 +185,33 @@ def default(session, install_grpc=True, prerelease=False, install_async_rest=Fal
         "python",
         "-m",
         "pytest",
-        *(
-            # Helpful for running a single test or testfile.
-            session.posargs
-            or [
-                "--quiet",
-                "--cov=google.api_core",
-                "--cov=tests.unit",
-                "--cov-append",
-                "--cov-config=.coveragerc",
-                "--cov-report=",
-                "--cov-fail-under=0",
-                # Running individual tests with parallelism enabled is usually not helpful.
-                "-n=auto",
-                os.path.join("tests", "unit"),
-            ]
-        ),
     ]
+    pytest_args.extend(["-W", "ignore::FutureWarning"])
+
+    pytest_args.extend(
+        [
+            # We use filterwarnings to ignore warnings that are out of our control,
+            # but we want to make sure that our own code does not generate warnings.
+            "-m",
+            "not filterwarnings",
```

Comment:
We do want to see warnings as that may require code changes. We can filter warnings here with a comment that links to a bug with more information https://github.com/googleapis/python-api-core/blob/628003e217d9a881d24f3316aecfd48c244a73f0/pyproject.toml#L91